test/inference.jl

```diff
@@ -276,4 +276,11 @@ let
     b = @code_lowered 1 + 1
     @test isa(a, LambdaInfo)
     @test isa(b, LambdaInfo)
+
+    Base.view(A::AbstractArray, i::Real) = view(A, :, i)
```

Comment:
could we make up a new local function here as opposed to extending an important base function for this test?